nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/provider/flow/git/GitFlowMetaData.java

```diff
@@ -136,36 +136,50 @@ private Repository openRepository(final File gitProjectRootDir) throws IOExcepti
     }
 
     private static boolean hasAtLeastOneReference(Repository repo) {
-        logger.info(repo.toString());
+        logger.info("Checking references for repository {}", repo.toString());
         for (Ref ref : repo.getAllRefs().values()) {
-            if (ref.getObjectId() == null)
+            if (ref.getObjectId() == null) {
                 continue;
+            }
             return true;
         }
         return false;
     }
 
+    /**
+     * Check if provided local repository exists or not at given 'Flow Storage Directory' in providers.xml
```

Comment:
Better:

Check if the provided local repository exists or not, provided by the 'Flow Storage Directory' configuration in the providers.xml.